transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/internal/LogEvent.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.android.datatransport.cct.internal;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class LogEvent {
+
+  public abstract long getEventTimeMs();
+
+  public abstract int getEventCode();
+
+  public abstract long getEventUptimeMs();
+
+  @SuppressWarnings("mutable")
+  @Nullable
+  public abstract byte[] getSourceExtension();
+
+  @SuppressWarnings("mutable")
+  @Nullable
+  public abstract byte[] getSourceExtensionJsonProto3Bytes();
+
+  public abstract long getTimezoneOffsetSeconds();
+
+  @Nullable
+  public abstract NetworkConnectionInfo getNetworkConnectionInfo();
+
+  @NonNull
+  public static Builder builder() {
+    return new AutoValue_LogEvent.Builder().setEventCode(0);
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    @NonNull
+    public abstract Builder setEventTimeMs(long value);
+
+    @NonNull
+    public abstract Builder setEventCode(int value);
+
+    @NonNull
+    public abstract Builder setEventUptimeMs(long value);
+
+    @NonNull
+    public abstract Builder setSourceExtension(@Nullable byte[] value);
+
+    @NonNull
+    public abstract Builder setSourceExtensionJsonProto3Bytes(@Nullable byte[] value);
```

Comment:
Only one of json or proto field can be set, can we make it impossile to set both? i.e. by having multiple builders(`jsonBuilder(byte[])`, `protoBuilder(byte[])`) and making setters package-private

Reply:
Done.